### ISSUE TITLE
Perhaps follow the W3C JSON forms spec?

### ISSUE THREAD

**Issue by dougwilson** [ljharb/qs#63]
It seems there is a technical report on adding in JSON to HTML forms: http://www.w3.org/TR/html-json-forms/

The syntax seems to pretty much be exactly what is being done here, but with slight differences (really, the differences just define the edge cases like how `foo=bar&foo[0]=baz` should parse: `{"foo":{"":"bar","0":"baz"}}`). It would also mean you have a good bullet against all the arguments that show up now and again for how this module operates :)

**Comment by nlf** [comment]
Yes! I'm so happy there's a standard that I can point to for this. Definitely going to do this, thank you for bringing this to my attention!

**Comment by dougwilson** [comment]
Cool :) The best part is it's nearly the same as this module already :) Feel free to back pedal on this issue as well; I haven't fully read the spec, and of course, you'll form better opinions on it once you do, haha.

**Comment by nlf** [comment]
Yeah, I'll read through the spec first, but I'm super excited if this will work. All the ambiguity in this module is a huge annoyance.

**Comment by dougwilson** [comment]
Plus all the bikeshedding regarding how it works :)

**Comment by nlf** [comment]
While I'd like to do this, I think this module isn't the right place for it. There are so many wild edge cases that this module already caters to that dropping them all in favor of this standard would likely cause adoption issues that I'd like to avoid. I'm going to close this for now, but potentially this could become a new module in the future.

**Comment by dougwilson** [comment]
Thanks for your thoughts, @nlf :) We'll write our own parser, then :)

**Comment by ljharb** [comment]
I'm happy to consider doing this in a semver-major bump - reopening, per https://github.com/expressjs/express/issues/3039#issuecomment-236029362
